internal/cli/uninstall.go

```diff
@@ -0,0 +1,203 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/clisnapshot"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	"github.com/hashicorp/waypoint/internal/serverinstall"
+)
+
+type UninstallCommand struct {
+	*baseCommand
+
+	platform         string
+	snapshotFilename string
+	skipSnapshot     bool
+	autoApprove      bool
+	deleteContext    bool
+}
+
+func (c *UninstallCommand) Run(args []string) int {
+	ctx := c.Ctx
+	log := c.Log.Named("uninstall")
+	defer c.Close()
+
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+
+	if !c.autoApprove {
+		c.ui.Output(strings.TrimSpace(autoApproveMsg), terminal.WithErrorStyle())
+		return 1
+	}
+
+	sg := c.ui.StepGroup()
+	defer sg.Wait()
+
+	// Pre-uninstall work
+	// - name the context we'll be uninstalling
+	// - generate a snapshot of the current install
+	s := sg.Add("")
+	defer func() { s.Abort() }()
+
+	contextDefault, err := c.contextStorage.Default()
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+	s.Update("Uninstalling Waypoint server with context %q", contextDefault)
+	s.Done()
+
+	// Generate a snapshot
+	s = sg.Add("")
+	if !c.skipSnapshot {
+		s.Update("Generating server snapshot...")
+		w, err := os.Create(c.snapshotFilename)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error creating snapshot file: %s", err)
+			return 1
+		}
+		if err = clisnapshot.WriteSnapshot(ctx, c.project.Client(), w); err != nil {
+			fmt.Fprintf(os.Stderr, "Error generating snapshot: %s", err)
+			return 1
+		}
+		s.Update("Snapshot %q generated", c.snapshotFilename)
+	} else {
+		s.Update("skip-snapshot set; not generating server snapshot")
+		s.Status(terminal.StatusWarn)
+	}
+	s.Done()
+
+	// Uninstall
+	p, ok := serverinstall.Platforms[strings.ToLower(c.platform)]
+	if !ok {
+		c.ui.Output(
+			"Error uninstalling server from %s: invalid platform",
+			c.platform,
+			terminal.WithErrorStyle(),
+		)
+
+		return 1
+	}
+
+	err = p.Uninstall(ctx, &serverinstall.InstallOpts{
+		Log: log,
+		UI:  c.ui,
+	})
+	if err != nil {
+		// point to current docs on manual server cleanup
```

Comment:
Maybe a missing TODO here :smile: 

```suggestion
		// TODO: point to current docs on manual server cleanup
```

Reply:
Ah! I can go ahead and complete that todo now since we have our Troubleshooting page. Thanks for catching this!